tokio/src/io/async_fd.rs

```diff
@@ -243,6 +243,25 @@ impl<T: AsRawFd> AsyncFd<T> {
         Self::new_with_handle_and_interest(inner, scheduler::Handle::current(), interest)
     }
 
+    /// Create a new AsyncFd with the provided raw epoll flags for registration.
+    ///
+    /// These flags replace any epoll flags would normally set when registering the fd.
+    ///
+    ///  **Note**: This is an [unstable API][unstable]. The public API of this may break in 1.x
+    /// releases.
+    /// See [the documentation on unstable features][unstable] for details.
+    ///
+    ///  [unstable]: crate#unstable-features
```

Comment:
Please add to the documentation that `EPOLLONESHOT` must not be used, and that `EPOLLET` must be set. And please add debug asserts for this.

Reply:
Done.